src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1093,7 +1093,17 @@ void Arguments::print_jvm_flags_on(outputStream* st) {
 void Arguments::print_jvm_args_on(outputStream* st) {
   if (_num_jvm_args > 0) {
     for (int i=0; i < _num_jvm_args; i++) {
-      st->print("%s ", _jvm_args_array[i]);
+      // quote any argument containing whitespaces
+      const char* const s = _jvm_args_array[i];
+      bool ws = false;
+      for (const char* p = s; *p && !ws; p++) {
+        ws = isspace(*p);
```

Comment:
Based on a [recommendation](https://en.cppreference.com/w/cpp/string/byte/isspace#:~:text=character%2C%20zero%20otherwise.-,Notes,-Like%20all%20other) from CPPReference, the argument of `isspace` should be casted to `unsigned char`.

Reply:
Note that this is not `std::isspace`. So, CPP ref does not directly apply here. It's the [Posix isspace](https://pubs.opengroup.org/onlinepubs/007904975/functions/isspace.html).

Reply:
ok, done.